cfme/storage/volume.py

```diff
@@ -206,6 +218,39 @@ def create_backup(self, name, incremental=None, force=None):
                  timeout=1000,
                  fail_func=self.refresh)
 
+    def create_snapshot(self, name, cancel=False, reset=False):
+        """create snapshot of cloud volume"""
+        initial_snapshot_count = self.snapshots_count
+        view = navigate_to(self, 'Snapshot')
+
+        changed = view.snapshot_name.fill(name)
+
+        # For changes only Save and Reset button activate
+        if changed:
+            if reset:
+                view.reset.click()
+                view.flash.assert_message('All changes have been reset')
+
+            elif cancel:
+                view.cancel.click()
+                view = self.create_view(VolumeDetailsView)
+                view.wait_displayed(timeout='10s')
+                view.flash.assert_message(
+                    'Snapshot of Cloud Volume "{}" was cancelled by the user'.format(self.name))
+
+            else:
+                view.save.click()
+
+                view = self.create_view(VolumeDetailsView)
+                view.wait_displayed(timeout='10s')
+                view.flash.assert_success_message('Snapshot for Cloud Volume "{}" '
```

Comment:
**Required**

Move this flash message assertion to the test. This prevents a changed flash message from blocking tests that need to create one for further testing.

Since you've got a new test here with a fixture, it should be simple to create a view (BaseLoggedInPage has flash and would be appropriate) and assert this success message there.

Here, just use view.flash.assert_no_error() to at least make sure that the action was successful - just not asserting the specific flash message text.

Reply:
@mshriver: I agreed with your suggestion l.e. changes in the flash message will avoid blocking further testing. But while creating `snapshot` or `backup` for `storage volume` I need a check for `snapshot count`. If I assert this message in test it will disappear after `wait_for` specific count.

Reply:
Noted, and I think that wait can also be removed from this CRUD method and moved to the test as well.

I think a good way to do this would be for this method to instantiate and return a VolumeSnapshot instance on success (None on cancel/failure).

Then any framework components using it further, or tests, can simply use `Volume.snapshots_count` before/after create to validate counts, and can use the returned instance to assert it exists.